components/movementsensor/gpsnmea/serial.go

```diff
@@ -144,14 +146,36 @@ func (g *SerialNMEAMovementSensor) GetCorrectionInfo() (string, uint) {
 	return g.correctionPath, g.correctionBaudRate
 }
 
+//nolint
 // Position position, altitide.
 func (g *SerialNMEAMovementSensor) Position(ctx context.Context, extra map[string]interface{}) (*geo.Point, float64, error) {
+	lastPosition := g.lastposition.GetLastPosition()
+
 	g.mu.RLock()
 	defer g.mu.RUnlock()
-	if g.data.location == nil {
-		return geo.NewPoint(0, 0), 0, errNilLocation
+
+	currentPosition := g.data.location
+
+	if currentPosition == nil {
+		return lastPosition, 0, errNilLocation
+	}
+
+	// if current position is (0,0) we will return the last non zero position
+	if g.lastposition.IsZeroPosition(currentPosition) && !g.lastposition.IsZeroPosition(lastPosition) {
+		return lastPosition, g.data.alt, g.err.Get()
 	}
-	return g.data.location, g.data.alt, g.err.Get()
+
+	// updating lastposition if it is different from the current position
+	if !g.lastposition.ArePointsEqual(currentPosition, lastPosition) {
+		g.lastposition.SetLastPosition(currentPosition)
+	}
+
+	// updating the last known valid position if the current position is non-zero
+	if !g.lastposition.IsZeroPosition(currentPosition) && !g.lastposition.IsPositionNaN(currentPosition) {
```

Comment:
[q] what if I'm near 0,0 globally? off the coast of west Africa?

Reply:
IsZeroPosition is only there to set last position. As long as currentPosition is not (0,0) we will update last position. So if current position is (0.001, 0.02), last position will be updated. I am open to suggestions on how to handle this since I have decided to avoid (0,0) altogether. But this doesn't interfere with the actual position of the gps.